### PR TITLE
JENKINS-29170 - `Util.fixEmpty` `charset` and `mimeType`

### PR DESCRIPTION
**Pull request by Evildethow**
[JENKINS-29170](https://issues.jenkins-ci.org/browse/JENKINS-29170)

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/steps/MailStep.java [hunk 1]
```diff
@@ -145,10 +146,10 @@ private MimeMessage buildMimeMessage() throws UnsupportedEncodingException, Mess
             if (step.bcc != null) {
                 messageBuilder.addRecipients(step.bcc, Message.RecipientType.BCC);
             }
-            if (step.charset != null) {
+            if (Util.fixEmpty(step.charset) != null) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/MailStep.java]: :bug: do not do this here. Rather do it in the setters. Read the [guide](https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#handling-default-values).

**Comment by Evildethow** [comment]
@jglick Thanks for the pointer. Moved into setters to support the snippet generator

File: src/main/java/org/jenkinsci/plugins/workflow/steps/MailStep.java [hunk 2]
```diff
@@ -50,7 +51,6 @@
  */
 public class MailStep extends AbstractStepImpl {
 
-    @DataBoundSetter
     public String charset;
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/MailStep.java]: May as well make them `private` while you are at it.

**Comment by jglick** [comment]
:bee:

**Comment by jglick** [comment]
re:bee:

**Pull request merged**